cmd/snapshotgc/snapshotgc.go

```diff
@@ -57,3 +57,33 @@ func getSnapshotsForNSReleases(
 	}
 	return snapToData, nil
 }
+
+// Gets a map to allow to tell with direct lookup if a snapshot is associated with
+// a SnapshotEnvironmentBinding resource
+func getSnapshotsForNSBindings(
+	cl client.Client,
+	snapToData map[string]snapshotData,
+	namespace string,
+	logger logr.Logger,
+) (map[string]snapshotData, error) {
+	binds := &applicationapiv1alpha1.SnapshotEnvironmentBindingList{}
+	err := cl.List(
+		context.Background(),
+		binds,
+		&client.ListOptions{Namespace: namespace},
+	)
+	if err != nil {
+		logger.Error(err, "Failed to list bindings")
+		return nil, err
+	}
+
+	for _, bind := range binds.Items {
+		data, ok := snapToData[bind.Spec.Snapshot]
+		if !ok {
+			data = snapshotData{}
+		}
+		data.environmentBinding = bind
+		snapToData[bind.Spec.Snapshot] = data
+	}
+	return snapToData, nil
```

Comment:
Actually we don't have to return `snapToData` it's passed by reference so we don't have to return the same ref as is in param.

Reply:
I don't like the practice of relying on side effects. I prefer returning the result directly. It also makes testing more straightforward.

Reply:
What side effects? Reference/pointer parameters are well documented base concept of golang. 
You must hate go-client that uses only references to update objects.

What is no nice with such function is, that because developer received a return value, the one may expect that the original value is unchanged and still work with it as independent copy, but that's not true, it's the same reference.

I don't get how unit-testing is more straightforward.

Reply:
> You must hate go-client that uses only references to update objects.

Indeed I do!

> What is no nice with such function is, that because developer received a return value, the one may expect that the original value is unchanged and still work with it as independent copy, but that's not true, it's the same reference.

That is indeed a downside.

> I don't get how unit-testing is more straightforward.

It's much easier to mock something that just returns a value instead of modifying its inputs.

Reply:
ah mocking is always fun

as I said, not a merge blocker for me